pkg/healthcheck/healthcheck.go

```diff
@@ -307,7 +311,14 @@ func (hc *HealthChecker) allCategories() []category {
 					description: "can create CustomResourceDefinitions",
 					hintAnchor:  "pre-k8s-cluster-k8s",
 					check: func(context.Context) error {
-						return hc.checkCanCreate(hc.ControlPlaneNamespace, "apiextensions.k8s.io", "v1beta1", "CustomResourceDefinition")
+						return hc.checkCanCreate("", "apiextensions.k8s.io", "v1beta1", "CustomResourceDefinition")
+					},
+				},
+				{
+					description: "has NET_ADMIN capability",
```

Comment:
If linkerd is installed with linkerd-cni enabled, then this check isn't required. How would you feel about adding an additional `linkerd check` flag for `--linkerd-cni-enabled`, and then only executing this check when the flag is unset? I'd expect it to work pretty similarly to check's `--single-namespace` flag.